### PR TITLE
Android obfuscation

### PR DESCRIPTION
**Pull request by swavkulinski**
Basic Android obfuscation and minification.

As suggested in https://github.com/flutter/flutter/pull/16956

### CODE REVIEW

File: android-release-build.md [hunk 1]
```diff
@@ -139,6 +139,54 @@ file.
...
+## Minfify and obfuscate
+
+By default, Flutter does not obfuscate or minify Android wrapper. If you intend to use third-party Java or Android libraries, you may want to reduce the size of the APK or protect that code from reverse engineering.
+
+### Step 1 - Configure obfuscation tool
+
+Create `/android/app/proguard-rules.pro` file and add rules listed below.
+
+```
+#Flutter Wrapper
```
Comment (eseidelGoogle) [inline android-release-build.md]: Should this just be part of our template?  @tvolkert @mravn-google ?

Comment (swavkulinski) [reply android-release-build.md]: @eseidelGoogle this was the initial idea, however I agree that obfuscation and shrinking is only needed in specific scenarios e.g. hybrid app with large libraries, non-feature requirements etc.

Also running ProGuard slows down build time, which maybe not the big issue for Flutter developers but has it's impact on CI build times.

**Comment by mravn-google** [comment]
I'd prefer to keep it out of the template---at least until we have a template with a clearer separation of files depended on by Flutter tooling vs those that are for the app author to provide/modify (this is work in progress).

**Comment by Sfshaza** [comment]
If we're not putting it into the template, I can add it here. Yes, @mravn-google @eseidelGoogle  ?

Unless I hear otherwise, I'll merge this in the next few.

**Comment by mravn-google** [comment]
@Sfshaza sgtm

**Comment by Sfshaza** [comment]
Thanks @swavkulinski!!

**Pull request merged**